### ISSUE TITLE
Add general SGD classifier with various loss functions and regularization

### ISSUE THREAD

**Issue by myui** [myui/hivemall#14]
- Support lasso+ridge+elasticnet regularization

http://scikit-learn.org/stable/modules/sgd.html#implementation-details
http://scikit-learn.org/stable/modules/linear_model.html#generalized-linear-models

https://github.com/yinlou/mltk/tree/master/src/mltk/predictor/glm
- SGD+L1 regularization

Stochastic Gradient Descent Training for L1-regularized Log-linear Models with Cumulative Penalty
Yoshimasa Tsuruoka, Jun’ichi Tsujii and Sophia Ananiadou, ACL2009.
http://aclweb.org/anthology/P/P09/P09-1054.pdf
http://blog.unnono.net/2010/02/l1-sgd.html (in Japanese)
- FOBOS (online learning + L1 regularization)

http://www.cs.berkeley.edu/~jduchi/projects/DuchiSi09_folos.html
http://d.hatena.ne.jp/Goto_youthK/20120215/1329234331 (in Japanese)
